### PR TITLE
iio demo updates

### PR DESCRIPTION
**Pull request by dbogdan**

### CODE REVIEW

**Comment by dbogdan** [comment]
v2:
 - Fix astyle complaints

File: drivers/adc/adc_demo/adc_demo.c [hunk 1]
```diff
@@ -111,6 +112,12 @@ int32_t adc_demo_remove(struct adc_demo_desc *desc)
 	return SUCCESS;
 }
 
+/**
+ * @brief active adc channels
+ * @param desc - descriptor for the adc
```
Comment (amiclaus) [inline drivers/adc/adc_demo/adc_demo.c:117]: one minor nit: `desc` -> `dev` (this will fix the doxygen build)

**Review by amiclaus — APPROVED** [review]
lgtm, one nit regarding documentation

**Comment by dbogdan** [comment]
v3:
 - Fix doxygen complaints

**Pull request merged**